test_contextual_lenses_learning.py

```diff
@@ -0,0 +1,629 @@
+# -*- coding: utf-8 -*-
+"""test_contextual_lenses_learning.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1PykmTCyuETlX4UFqoFCRpAJLIAxbAyg2
+
+# Imports
+"""
+
+import functools
+import itertools
+import os
+import time
+
+import flax
+from flax import jax_utils
+from flax import nn
+from flax import optim
+
+import jax
+from jax import random
+from jax import lax
+import jax.nn
+import jax.numpy as jnp
+
+import numpy as np
+
+import pandas as pd
+
+import matplotlib.pyplot as plt
+
+import scipy.stats
+
+import sklearn.linear_model
+import sklearn.metrics
+
+import seaborn as sns
+
+"""# Contextual lenses"""
+
+def max_pool(x, rep_size=None):
+  # Take maximum over sequence length
+
+  rep = jnp.max(x, axis=-2)
+   
+  return rep
+
+
+def mean_pool(x, rep_size=None):
+  # Take mean over sequence length
+
+  rep = jnp.mean(x, axis=-2)
+  
+  return rep
+
+
+def linear_max_pool(x, rep_size=256):
+  # Apply linear transformation + ReLU
+  # Take maximum over sequence length
+  
+  x = nn.Dense(
+        x,
+        rep_size,
+        kernel_init=nn.initializers.xavier_uniform(),
+        bias_init=nn.initializers.normal(stddev=1e-6))
+  
+  x = nn.relu(x)
+  
+  rep = max_pool(x, rep_size)
+  
+  return rep
+
+
+def linear_mean_pool(x, rep_size=256):
+  # Apply linear transformation + ReLU
+  # Take mean over sequence length
+  
+  x = nn.Dense(
+        x,
+        rep_size,
+        kernel_init=nn.initializers.xavier_uniform(),
+        bias_init=nn.initializers.normal(stddev=1e-6))
+  
+  x = nn.relu(x)
+  
+  rep = mean_pool(x, rep_size)
+  
+  return rep
+
+"""# Models"""
+
+def create_optimizer(model, learning_rate, weight_decay):
```

Comment:
move all of the stuff that's not for contextual lenses to a different file. Could call this train.py or util.py or something.